IoTCIntegration/lib/engine.js

```diff
@@ -39,12 +39,22 @@ module.exports = async function (context, device, measurements) {
         throw new StatusError('Invalid format: invalid measurement list.', 400);
     }
 
+    if (timestamp && isNaN(Date.parse(timestamp))) {
```

Comment:
if someone passes `''` or `0` for timestamp will that cause problems?

i'd probably tighten the first part of the check to be safe

```suggestion
    if (timestamp != null && isNaN(Date.parse(timestamp))) {
```

Reply:
ah okay i see the condition below. consider this optional depending on whether you want to tell the user that falsy values won't do anything